src/main/resources/JavadocLexer.g4

```diff
@@ -182,11 +183,13 @@ Param_WS: [ \t]+ -> type(WS), channel(WHITESPACES);
 
 mode tag;
 
-TAG_CLOSE: '>' -> popMode;
-TAG_SLASH_CLOSE: '/>' -> popMode;
+TAG_CLOSE: '>' {hasSeenTagName = false;} -> popMode;
+TAG_SLASH_CLOSE: '/>' {hasSeenTagName = false;} -> popMode;
 TAG_SLASH: '/';
 TAG_EQUALS: '=' -> pushMode(attrValue);
-TAG_NAME: TagNameStartChar TagNameChar*;
+// TODO: this can be one rule maybe?
+TAG_NAME: {hasSeenTagName == false}? TagNameStartChar TagNameChar* {hasSeenTagName = true;};
+TAG_ATTR_NAME: {hasSeenTagName == true}? TagNameStartChar TagNameChar*;
```

Comment:
There are lots of ways to do this, but this was the first to come to mind. We need to differentiate between these tokens.